dora/core/client/fs/src/main/java/alluxio/client/file/dora/ConsistentHashProvider.java

```diff
@@ -128,21 +131,24 @@ public List<BlockWorkerInfo> getMultiple(String key, int count) {
    * update the state of the hash provider using the worker list provided by that thread, and all
    * others will not change the internal state of the hash provider.
    *
-   * @param workerInfos the up-to-date worker list
+   * @param workers         the up-to-date worker list
    * @param numVirtualNodes the number of virtual nodes used by consistent hashing
    */
-  public void refresh(List<BlockWorkerInfo> workerInfos, int numVirtualNodes) {
-    Preconditions.checkArgument(!workerInfos.isEmpty(),
-            "cannot refresh hash provider with empty worker list");
-    maybeInitialize(workerInfos, numVirtualNodes);
+  public void refresh(List<WorkerIdentity> workers, int numVirtualNodes) {
+    Preconditions.checkArgument(!workers.isEmpty(),
+        "cannot refresh hash provider with empty worker list");
+    maybeInitialize(workers, numVirtualNodes);
     // check if the worker list has expired
     if (shouldRebuildActiveNodesMapExclusively()) {
       // thread safety is valid provided that build() takes less than
       // WORKER_INFO_UPDATE_INTERVAL_NS, so that before next update the current update has been
       // finished
-      if (hasWorkerListChanged(workerInfos, mLastWorkerInfos.get())) {
-        mActiveNodesByConsistentHashing = build(workerInfos, numVirtualNodes);
-        mLastWorkerInfos.set(workerInfos);
+      List<WorkerIdentity> lastWorkerIds = mLastWorkers.get();
+      if (!workers.equals(lastWorkerIds)) {
```

Comment:
A `List.equals()` is `O(n)`. However this is not more complex than the existing `hasWorkerListChanged()` so this is okay

Reply:
however watch out `List.equals()` returns false if the two lists are of different order but same instances

Reply:
should change to `Set<WorkerIdentity>` instead.

Reply:
changed to set instead of list.